### ISSUE TITLE
DartPad autocomplete doesn't always fit on screen.

### ISSUE THREAD

**Issue by mmcdon20** [dart-lang/dart-pad#2122]
### Describe the problem

The autocomplete menu doesn't always fit entirely on the screen making it difficult to read the available options.

This happens when the text for an autocomplete option is too long to fit within the available space. 

### Expected fix

Update the autocomplete menu to fit on screen better.

### Additional context

As an example, if you enter the following code on DartPad:

```dart
import 'dart:ui' as ui;

void main() {
  ui.
}
```

You will see an autocomplete menu that looks like this:

![autocomplete](https://user-images.githubusercontent.com/5104543/147625866-33f59b39-d9ad-4401-acca-f09be91ce135.png)

**Comment by RedBrogdon** [comment]
IIRC, the autocompletes are done through CodeMirror's UI:

https://github.com/dart-lang/dart-pad/blob/master/lib/editing/editor_codemirror.dart

That could make it difficult to adjust the HTML/CSS, unless CodeMirror offers a means to do it through their API. We might have to manually truncate strings before handing them over.

Either way, this is something that's legit annoying and should get fixed.

**Comment by mmcdon20** [comment]
@RedBrogdon from inspecting the elements in the browser, it looks like the relevant css classes are `.CodeMirror-hints` for the completion menu, and `.CodeMirror-hint` for each item within the menu. 

I inserted the following css into chrome devtools, and the behavior was a lot more reasonable:

```css
/* Code completion menu */
.CodeMirror-hints {
  /* or whatever width limit */
  max-width: min(80%, 800px);
}

/* Item in the code completion menu */
.CodeMirror-hint {
  /* show ellipsis on overflow */
  padding-right: 5px !important;
  list-style-position: inside;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;   
}

/* A mouse hover color might be nice to add too */
.CodeMirror-hint:hover:not(.CodeMirror-hint-active) {
  background-color: aliceblue;
}
```

With the above css, I got these results:

![dart-pad-css](https://user-images.githubusercontent.com/5104543/148337240-8af19d3b-adc0-4636-ae90-70d88e56436f.png)

**Comment by mmcdon20** [comment]
This was resolved with #2128. Closing this issue.
